### PR TITLE
Use wide-string API instead of SetConsoleOutputCP.

### LINKED ISSUE #1121: Don't call SetConsoleOutputCP on windows
https://github.com/stedolan/jq/commit/8aecf82bfef6647315eb96e69eac3f65ce38ee70#diff-2045016cb90d1e65d71c2407a2570927R180

Stop calling this function. Calling this function change window size of cmd.exe . There are some environment that have problem with chaning console code page. For example, japanese.

Before calling
http://go-gyazo.appspot.com/16c3eb9448d7fc17.png

After calling
http://go-gyazo.appspot.com/17bffeccb0a03d29.png

And you don't restore original code page.

### PR DESCRIPTION
**Pull request by mattn**
Fixes #1121 

Related Issue #824 #262

### CODE REVIEW

**Comment by mattn** [comment]
ping

**Comment by mattn** [comment]
ping again

**Comment by nicowilliams** [comment]
Hi.  Thanks for this wonderful contribution!  I'll merge it this weekend.

**Comment by mattn** [comment]
ping

File: src/jv_print.c [hunk 1]
```diff
@@ -33,9 +33,18 @@ static void put_buf(const char *s, int len, FILE *fout, jv *strout, int is_tty)
...
   /* See util.h */
-  if (is_tty)
-    WriteFile((HANDLE)_get_osfhandle(fileno(fout)), s, len, NULL, NULL);
-  else
+  if (is_tty) {
+    wchar_t* ws;
+    size_t wl;
+    if (len == -1) len = strlen(s);
+    wl = MultiByteToWideChar(CP_UTF8, 0, s, len,  NULL, 0);
+    ws = (wchar_t*) malloc((wl + 1) * sizeof(wchar_t));
```
Comment (nicowilliams) [inline src/jv_print.c]: No need to cast the result of malloc() :)

File: configure.ac [hunk 2]
```diff
@@ -214,6 +214,14 @@ AC_C_BIGENDIAN(
 
 AC_SUBST([BUNDLER], ["$bundle_cmd"])
 
+case "$host" in
+    *mingw*)
+        LIBS="$LIBS -lshlwapi"
```
Comment (nicowilliams) [inline configure.ac]: In Makefile.am we have this:

```
if WIN32
libjq_la_LIBADD += -lshlwapi
endif
```

Ah, but we need that for jq too, not just libjq.  I think I'd rather change Makefile.am than configure.ac for this.

File: src/jv_file.c [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 jv jv_load_file(const char* filename, int raw) {
   FILE* file = fopen(filename, "r");
-  struct jv_parser* parser;
+  struct jv_parser* parser = NULL;
```
Comment (nicowilliams) [inline src/jv_file.c]: Ah, yes, some compilers are dumb and will emit a warning without this...

File: src/main.c [hunk 4]
```diff
@@ -177,7 +177,6 @@ int main(int argc, char* argv[]) {
   jv program_arguments = jv_array();
 
 #ifdef WIN32
-  SetConsoleOutputCP(CP_UTF8);
```
Comment (nicowilliams) [inline src/main.c]: Is this something we should generally do if `isatty(STDOUT_FILENO)`, or should we never do this?

**Comment by nicowilliams** [comment]
I'll modify this a bit and if it still builds in Appveyor, I'll push.

**Comment by nicowilliams** [comment]
Well, I can't get `./configure` to get past looking for `flex` on Appveyor.  I tried debugging with RDP, but I couldn't connect.  I've opened [#1346 against Appveyor](https://github.com/appveyor/ci/issues/1346).

**Comment by nicowilliams** [comment]
Getting closer...

**Comment by nicowilliams** [comment]
Pushed this as:

```
2e1f2db Improve Appveyor build
ba32b33 Windows: configure fails to detect missing math
6e77b01 Make comment consistent with code
5d058fa Don't call SetConsoleOutputCP
12c60b2 Quiet a compiler warning
9801de6 Windows: Also use -lshlwapi for jq(1)
```

with slight changes.

Thanks!

**Comment by nicowilliams** [comment]
The Appveyor build should be passing soon.

**Comment by mattn** [comment]
Could you please make new release include this change?

**Comment by mattn** [comment]
please make new release. I still have this issue. You can confirm following gif that jq change console codepage.

terminal6

**Pull request closed without merging**